wgengine/magicsock/magicsock.go

```diff
@@ -4195,6 +4196,21 @@ func (de *endpoint) updateFromNode(n *tailcfg.Node, heartbeatDisabled bool) {
 	de.sendFunc.Store(nil)
 }
 
+// isDuplicatePing records the txID for the endpoint. It returns true if
```

Comment:
Go style nit: s/returns true if/reports whether/

Also: this name sounds like it wouldn't have a side effect of recording the last value. Not that I have a better suggestion.

Reply:
This function was removed in favor of plumbing into `addCandidateEndpoint()`, which uses `reports` instead of `returns` in its doc.